### ISSUE TITLE
Machine Learning not working on Nvidia GPU

### ISSUE THREAD

**Issue by VijayakumarRavi** [immich-app/immich#14519]
### The bug

smart search is not working on Nvidia GPU.

no matter what i do machine learning container always says
`FAIL : provider_options_utils.h:151 Parse Failed to parse provider option "device_id": CUDA failure 100: no CUDA-capable device is detected ; GPU=-1`

Possible duplicate of #10334

### The OS that Immich Server is running on

Ubuntu 24.04.1 LTS

### Version of Immich Server

v1.122.0

### Version of Immich Mobile App

v1.122.0

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
---
name: immich
services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:release
    volumes:
      - ./library:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file: [.env]
    environment:
      NVIDIA_VISIBLE_DEVICES: all
    depends_on: [redis, database]
    restart: unless-stopped
    labels:
      - traefik.enable=true
      - traefik.http.routers.immich-server.rule=Host(`photo.sanji.vijayakumar.xyz`)
      - traefik.http.routers.immich-server.entrypoints=https
      - traefik.http.routers.immich-server.tls=true
      - traefik.http.routers.immich-server.tls.certresolver=cloudflare
      - traefik.http.services.immich-server.loadbalancer.server.port=2283
    networks: [immich, proxy]
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities:
                - gpu
                - compute
                - video
  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:release-cuda
    volumes: [model-cache:/cache]
    env_file: [.env]
    environment:
      NVIDIA_VISIBLE_DEVICES: all
    restart: unless-stopped
    networks: [immich]
    devices:
      - /dev/nvidia-caps:/dev/nvidia-caps
      - /dev/nvidia0:/dev/nvidia0
      - /dev/nvidiactl:/dev/nvidiactl
      - /dev/nvidia-modeset:/dev/nvidia-modeset
      - /dev/nvidia-uvm:/dev/nvidia-uvm
      - /dev/nvidia-uvm-tools:/dev/nvidia-uvm-tools
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities:
                - gpu

  redis:
    container_name: immich_redis
    image: docker.io/redis:6.2-alpine
    healthcheck:
      test: redis-cli ping || exit 1
    restart: unless-stopped
    networks: [immich]
  database:
    container_name: immich_postgres
    image: registry.hub.docker.com/tensorchord/pgvecto-rs:pg14-v0.2.0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_INITDB_ARGS: --data-checksums
    volumes: [pgdata:/var/lib/postgresql/data]
    healthcheck:
      test: pg_isready --dbname='${DB_DATABASE_NAME}' || exit 1; Chksum="$$(psql --dbname='${DB_DATABASE_NAME}'
        --username='${DB_USERNAME}' --tuples-only --no-align --command='SELECT COALESCE(SUM(checksum_failures),
        0) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [
        "$$Chksum" = '0' ] || exit 1
      interval: 5m
      start_interval: 30s
      start_period: 5m
    command:
      - postgres
      - -c
      - shared_preload_libraries=vectors.so
      - -c
      - search_path="$$user", public, vectors
      - -c
      - logging_collector=on
      - -c
      - max_wal_size=2GB
      - -c
      - shared_buffers=512MB
      - -c
      - wal_compression=on
    restart: unless-stopped
    networks: [immich]
  backup:
    container_name: immich_db_dumper
    image: prodrigestivill/postgres-backup-local
    env_file: [.env]
    environment:
      POSTGRES_HOST: database
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
        #      SCHEDULE: "@daily"
      SCHEDULE: '@hourly'
      BACKUP_DIR: /db_dumps
      HC_URL: https://hc-ping.com/67265230-8a84-4741-ae16-e01edf023a97
    volumes: [./backup/db_dumps:/db_dumps, ./backup/hooks:/hooks]
    depends_on: [database]
    restart: unless-stopped
    networks: [immich]
volumes:
  model-cache:
  pgdata:
networks:
  immich:
    name: immich
  proxy:
    external: true
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=./library

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=********

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis

TZ=Asia/Kolkata
```

### Reproduction steps

1. Start immich docker stack
2. Wait for sometime
3. Try smart search job
...

### Relevant log output

```shell
immich_machine_learning  | 2024-12-06 11:20:09.799304248 [E:onnxruntime:Default, cuda_call.cc:118 CudaCall] CUDA failure 100: no CUDA-capable device is
detected ; GPU=-1 ; hostname=c4133a2e6885 ; file=/onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider_info.cc ; line=66 ; expr=cudaG
etDeviceCount(&num_devices);
immich_machine_learning  | *************** EP Error ***************
immich_machine_learning  | EP Error /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider_info.cc:59 static onnxruntime::CUDAExecutio
nProviderInfo onnxruntime::CUDAExecutionProviderInfo::FromProviderOptions(const onnxruntime::ProviderOptions&) [ONNXRuntimeError] : 1 : FAIL : provider_
options_utils.h:151 Parse Failed to parse provider option "device_id": CUDA failure 100: no CUDA-capable device is detected ; GPU=-1 ; hostname=c4133a2e
6885 ; file=/onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider_info.cc ; line=66 ; expr=cudaGetDeviceCount(&num_devices);
immich_machine_learning  |  when using ['CUDAExecutionProvider', 'CPUExecutionProvider']
immich_machine_learning  | Falling back to ['CPUExecutionProvider'] and retrying.
immich_machine_learning  | ****************************************
```

### Additional information

output of `python -c "import onnxruntime as ort;print(ort.get_device())"` inside machine learning container
```bash
root@c4133a2e6885:/usr/src/app# python -c "import onnxruntime as ort;print(ort.get_device())"
GPU
```

```bash
❯ nvidia-container-toolkit --version
NVIDIA Container Runtime Hook version 1.16.2
commit: a5a5833c14a15fd9c86bcece85d5ec6621b65652
```

**Comment by mertalev** [comment]
The `devices` section you added is unnecessary and the `NVIDIA_VISIBLE_DEVICES` env is also already set in the image.

What's the GPU model you're using?

**Comment by VijayakumarRavi** [comment]
I copied the cuda section from `hwaccel.ml.yml` to the devices. can you please suggest what i need to add there

My GPU model is `NVIDIA GeForce GT 710`

**Comment by ChronSyn** [comment]
According to official Nvidia docs, the GT-710 is not CUDA supported in either legacy (https://developer.nvidia.com/cuda-legacy-gpus) or current (https://developer.nvidia.com/cuda-gpus).

As details in the Immich ML docs (https://immich.app/docs/features/ml-hardware-acceleration), it requires CUDA support with support for compute capability >= 5.2.

TLDR: The card you are trying to use is not supportable by Immich ML.
